emv2/org.osate.xtext.aadl2.errormodel/src/org/osate/xtext/aadl2/errormodel/validation/ErrorModelValidator.java

```diff
@@ -974,10 +974,20 @@ private void checkTransitionTargetTypes(ErrorBehaviorTransition ebt) {
 			if (tt == null || tt.getTypeTokens().isEmpty()) {
 				return;
 			}
+			if (tt.getTypeTokens().size() > 1) {
+				error(ebt,
+						"Target state " + ebs.getName()
+								+ " should be associated with only a single error type instead of "
+								+ tt.getTypeTokens().size() + " error types");
+			}
 			TypeToken ebtargetTS = tt.getTypeTokens().get(0);
 			if (ebtargetTS == null) {
 				return;
 			}
+			if (ebtargetTS.getType().get(0) instanceof TypeSet) {
```

Comment:
Add null check: getType may return null if the error type name has a typo and cannot be resolved.
aybe check `! ... instanceof ErrorType` which makes it clearer what is expected.